File: 1st.cpp
#include <bits/stdc++.h>
#include <iostream>
#include <string.h>
using namespace std;
int recurs(int arr1[], int strt, int endd, int val)
{
    int k;
    k = sizeof(arr1) / 2;
    int len;
    len = sizeof(arr1);
    if (arr1[k] > val)
    {
        return recurs(arr1, 0, k - 1, val);
    }
    if (arr1[k] < val)
    {
        return recurs(arr1, k + 1, len, val);
    }
}
int main()
{
    int arr[] = {10, 20, 30, 40, 50};
    int x;
    cout << "enter the number";
    cin >> x;
    if (arr[2] == x)
    {
        cout << "it found at middle";
    }
    else
    {
        int t;
        t = recurs(arr, 0, 5, x);
        if (t == x)
        {
            cout << "it is matched";
        }
        else
            cout << "it is not matched";
    }

    return 0;
}

File: 1st.java
import java.util.*;
public
class Main
{
public
    static int recurs(int[] arrn, int start, int end, int val)
    {
        int k = arrn[(arrn.length) / 2], j, p;
        if (arrn[k] > val)
        {
            recurs(arrn[k] / 2, start, k / 2, val);
            return arrn[k];
        }
        else
        {
            recurs(arrn[k] / 2, k / 2, end, val);
            return arrn[k];
        }
    }
public
    static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);
        System.out.print("enter the value you wnat to find it ");
        int x = sc.nextInt();
        int flag = 0;
        int[] arr = {10, 20, 30, 40, 50};
        // for(int i=0;i<arr.length;i++){
        //     if(arr[i]==x){
        //         flag=1;
        //         break;
        //     }
        //     else
        //     flag =0;
        // }

        // if(flag==1){
        //     System.out.println("you got it");
        // }
        // else{
        //     System.out.println("you did not got it");
        // }
        int strt = 0, endd = arr.length;
        int[] arrr2 = {};

        if (x == arr[(arr.length) / 2])
        {
            System.out.println("value got at middle of the array");
        }
        else
        {
            int arr2[] = recurs(arr, strt, endd, x);
            System.out.println("you got it from recursiion");
        }
    }
}